lib/datadog/tracing/sampling/span/matcher.rb

```diff
@@ -54,6 +56,13 @@ def match?(span)
             end
           end
 
+          def ==(other)
```

Comment:
Is a span really the same as another span if their service and name are the same? For sampling purposes?

Can you explain the logic behind this a little more?

Reply:
This equality method referrers to the `Matcher`, not to spans.
If two matchers have the exact same instance variables, which is the only state they can have, they are the same.
Currently, the `name` and `service` matchers are all the instance variables a `Matcher` can have, so if two matchers have the same `name` and `service` they are effectively the same.